### PR TITLE
stack_using_two_queues

### PR DESCRIPTION
**Pull request by IndranjanaChatterjee**
### Describe your change:

Added the code for the implementation of stack using two queues

Fixes:#9987

* [x] Add an algorithm?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: data_structures/stacks/stack_with_two_queues.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+""" Implementing stack using two arrays"""
+
+
+class Stack(Generic[T]):  # Stack class to implement stack operations
+    def __init__(self) -> None:
+        self.insert_queue = deque()  # First Queue to be used for inserting
+        self.suffle_queue = deque()  # Second Queue to be used for suffling
+
+    def push(self, item: int) -> None:
+        self.insert_queue.append(item)  # Add items into the Queue
```
Comment (cclauss) [inline data_structures/stacks/stack_with_two_queues.py:14]: The code already says what the comment repeats.
```suggestion
        self.insert_queue.append(item)
```

File: data_structures/stacks/stack_with_two_queues.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        while self.suffle_queue:
+            self.insert_queue.append(
+                self.suffle_queue.popleft()
+            )  # Popping the elements
```
Comment (cclauss) [inline data_structures/stacks/stack_with_two_queues.py:18]: Let's not wrap lines just to make room for a comment.
```suggestion
            # Popping the elements
            self.insert_queue.append(self.suffle_queue.popleft())
```

File: data_structures/stacks/stack_with_two_queues.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+        self.insert_queue, self.suffle_queue = self.suffle_queue, self.insert_queue
+
+    def pop(self) -> int:
+        if not (self.suffle_queue):  # if the stack  is empty
```
Comment (cclauss) [inline data_structures/stacks/stack_with_two_queues.py:23]:
```suggestion
        if not self.suffle_queue:  # if the stack  is empty
```

File: data_structures/stacks/stack_with_two_queues.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    def top(self) -> int:
+        if not (self.suffle_queue):
+            return None
+        return self.suffle_queue[0]
+
+    def printing(self) -> None:
+        print(self.suffle_queue)
+
+    def size(self) -> int:
+        return len(self.suffle_queue)
```
Comment (cclauss) [inline data_structures/stacks/stack_with_two_queues.py:36]: These `magic methods` will allow us to just `print(stack)` and `len(stack)`.
```suggestion
    def __str__(self) -> str:
        return tuple(self.suffle_queue)

    def __len__(self) -> int:
        return len(self.suffle_queue)
```

File: data_structures/stacks/stack_with_two_queues.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+
+def test_stack() -> None:
+    s = Stack()  # Creating a stack in S
```
Comment (cclauss) [inline data_structures/stacks/stack_with_two_queues.py:40]: Single-letter variable names makes code look like it was written in the 1970's.
```suggestion
    stack = Stack()  # Create a empty stack
```

**Review by cclauss — CHANGES_REQUESTED** [review]
Looks cool...

File: data_structures/stacks/stack_with_two_queues.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+from typing import Generic, TypeVar
+
+T = TypeVar("T")
+""" Implementing stack using two arrays"""
+
+
+class Stack(Generic[T]):  # Stack class to implement stack operations
+    def __init__(self) -> None:
+        self.insert_queue = deque()  # First Queue to be used for inserting
+        self.suffle_queue = deque()  # Second Queue to be used for suffling
```
[the remark this answers is not in the document]
Comment (cclauss) [reply data_structures/stacks/stack_with_two_queues.py:11]: OPTIONAL: Should `Stack` be a dataclass?

**Comment by IndranjanaChatterjee** [comment]
@cclauss I have tried to make the required changes in the PR #10109 . Please review. Apologies for the inconveniences caused.

**Comment by cclauss** [comment]
Fixed in data_structures/stacks/stack_using_two_queues.py

**Pull request closed without merging**